### PR TITLE
Don't expand docs unlikely to be owned by teacher

### PR DESCRIPTION
**Pull request by twhitnah**
If a doc's sole owners are the viewer or a group of teachers as defined in Google classroom, we shouldn't auto-expand to show the number of revisions since it's unlikely it's relevant in those cases.

Hopefully we can find more proxies for student work through the API (at least in the Google classroom case)

Test Plan:
1. Confirm student work expands
2. Confirm teacher docs don't

### CODE REVIEW

File: api/google/utils.py [hunk 1]
```diff
@@ -128,6 +129,15 @@ def get_num_revisions_for_doc(google_drive_service: Resource, document_id: str)
         revisions = []
     return len(revisions)
 
+def get_owner_email_addresses_for_doc(google_drive_service: Resource, document_id: str) -> List[str]:
```
Comment (armanjaffer) [inline api/google/utils.py:132]: how's the latency on this? curious if requesting permissions slows down this. No need to measure it -- but curious if you think it's slow

File: api/main.py [hunk 2]
```diff
@@ -194,7 +194,18 @@ def plagiarism_score():
                     }
             case 'google_revisions':
                 google_drive_service = get_google_drive_service(__required_google_credentials())
+                viewer_email_address = __required_user().email
```
Comment (armanjaffer) [inline api/main.py:197]: Would you be able to document how this code works? What happens for non-google classroom users?

File: api/google/utils.py (hunk 1 above)
Comment (twhitnah) [reply api/google/utils.py:132]: Good question.  It looks like the additional DB call is 76ms and the Google round trip is 373ms.  While the experience is asynchronous and the teacher likely will still be visually processing the page load before the response is returned, I can see if I can find a way to batch these.

Current breakdown is: 750ms to get credentials and around 350 to get the number of revisions.  So this additional request isn't too big a deal, but if we can give ourself more buffer for other requests we might make, that would be good

**Comment by twhitnah** [comment]
Recreated this PR here:
https://github.com/briskedu/brisk-learning/pull/36

Batched up the request and cut the IO time from 750ms to 450ms.

**Pull request closed without merging**